src/couch_index/src/couch_index_server.erl

```diff
@@ -322,15 +348,50 @@ handle_db_event(<<"shards/", _/binary>> = DbName, {ddoc_updated, DDocId}, St) ->
 handle_db_event(DbName, {ddoc_updated, DDocId}, St) ->
     lists:foreach(
         fun({_DbName, {_DDocId, Sig}}) ->
-            case ets:lookup(?BY_SIG, {DbName, Sig}) of
+            case ets:lookup(St#st.by_sig, {DbName, Sig}) of
                 [{_, IndexPid}] ->
                     (catch gen_server:cast(IndexPid, ddoc_updated));
                 [] ->
                     ok
             end
         end,
-        ets:match_object(?BY_DB, {DbName, {DDocId, '$1'}})
+        ets:match_object(St#st.by_db, {DbName, {DDocId, '$1'}})
     ),
     {ok, St};
 handle_db_event(_DbName, _Event, St) ->
     {ok, St}.
+
+num_servers() ->
+    erlang:system_info(schedulers).
+
+server_name(Arg) ->
+    name("index_server", Arg).
+
+by_sig(Arg) ->
+    name("couchdb_indexes_by_sig", Arg).
+
+by_pid(Arg) ->
+    name("couchdb_indexes_by_pid", Arg).
+
+by_db(Arg) ->
+    name("couchdb_indexes_by_db", Arg).
+
+belongs(Arg, #st{} = St) ->
+    server_name(Arg) == St#st.server_name.
+
+name(BaseName, Sig) when is_list(Sig) ->
+    name(BaseName, ?l2b(Sig));
+name(BaseName, Sig) when is_binary(Sig) ->
+    N = 1 + erlang:phash2(Sig, num_servers()),
+    name(BaseName, N);
```

Comment:
Minor nit: Would it make sense to preserve the `Arg` argument name  instead of `Sig`, otherwise it may seem like it could be related to the view signature?

Reply:
oof good catch. It _was_ the view signature at one point.